addons/packages/oracle-cpi/1.22.1/bundle/config/provider-config.lib.txt

```diff
@@ -0,0 +1,46 @@
+#@ load("@ytt:data", "data")
+
+(@ def get_provider_config(values): -@)
+
+useInstancePrincipals: true
+
+# compartment configures Compartment within which the cluster resides.
+compartment: (@= values.compartment @)
+
+# vcn configures the Virtual Cloud Network (VCN) within which the cluster resides.
+vcn: (@= values.vcn @)
+
+loadBalancer:
+  # subnet1 configures one of two subnets to which load balancers will be added.
+  # OCI load balancers require two subnets to ensure high availability.
+  subnet1: (@= values.loadBalancer.subnet1 @)
+
+  # subnet2 configures the second of two subnets to which load balancers will be
+  # added. OCI load balancers require two subnets to ensure high availability.
+  subnet2: (@= values.loadBalancer.subnet1 @)
+
+  # SecurityListManagementMode configures how security lists are managed by the CCM.
+  # If you choose to have security lists managed by the CCM, ensure you have setup the following additional OCI policy:
+  # Allow dynamic-group [your dynamic group name] to manage security-lists in compartment [your compartment name]
+  #
+  #   "All" (default): Manage all required security list rules for load balancer services.
+  #   "Frontend":      Manage only security list rules for ingress to the load
+  #                    balancer. Requires that the user has setup a rule that
+  #                    allows inbound traffic to the appropriate ports for kube
+  #                    proxy health port, node port ranges, and health check port ranges.
+  #                    E.g. 10.82.0.0/16 30000-32000.
+  #   "None":          Disables all security list management. Requires that the
+  #                    user has setup a rule that allows inbound traffic to the
+  #                    appropriate ports for kube proxy health port, node port
+  #                    ranges, and health check port ranges. E.g. 10.82.0.0/16 30000-32000.
+  #                    Additionally requires the user to mange rules to allow
+  #                    inbound traffic to load balancers.
+  securityListManagementMode: All
```

Comment:
This should be overridable

Reply:
Also, checking with Shyam, if the mode is all, we do also then have to set https://github.com/oracle/oci-cloud-controller-manager/blob/master/manifests/provider-config-example.yaml#L56-L58

Reply:
i've kept that noted. let's have a separate PR to address this